ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -5496,16 +5507,73 @@ NtFreeVirtualMemory(IN HANDLE ProcessHandle,
                 else
                 {
                     //
-                    // This is case B and the hardest one. Because we are removing
-                    // a chunk of memory from the very middle of the VAD, we must
-                    // actually split the VAD into two new VADs and compute the
-                    // commit charges for each of them, and reinsert new charges.
+                    // Case B (freeing a part in the middle)
+                    //
+                    // This is the hardest one. Because we are removing a chunk
+                    // of memory from the very middle of the VAD, we must actually
+                    // split the VAD into two new VADs and compute the commit
+                    // charges for each of them, and reinsert new charges.
+                    //
+                    NewVad = ExAllocatePoolZero(NonPagedPool, sizeof(MMVAD_LONG), 'SdaV');
```

Comment:
Nit: I'd rather move variable declarations to this block

Reply:
Not the prevalent style anywhere around here so I'd rather not.